liquidprompt

```diff
@@ -1057,6 +1189,28 @@ _lp_fill() {
     lp_fill="${left}${filled}${right}"
 }
 
+_lp_substitute() { # string pairs_array
+    # Replace string with the right part of the first pair in the given array, if the left part matches the string.
+    # Pairs should be separated by an `=` sign.
+    # FIXME If we ever support Bash>4.3, this would be better implemented with associative arrays.
+    lp_substitute=""
+    local string="${1}"
+    shift 1
+    local pairs=("$@")
+    [[ -n "${pairs[*]}" ]] || return 2
+
+    local key_rep key rep
+    for key_rep in "${pairs[@]}"; do
+        key="${key_rep%=*}"
```

Comment:
This will only work if the input strings have only a single `=`. If they have multiple, it will mangle. The easy fix is to change this to
```suggestion
        key="${key_rep%%=*}"
```
so that both expressions split on the first `=`. Obviously, if a user wants to replace a string containing a `=`, that will be impossible.

The other way to fix this is to have the pairs be passed in as separate arguments, like
`_lp_substitute "my long string" "long" "short"`

Reply:
I thought about the simple pair, but I failed to find an easy way to pass a whole list of replacements, instead of just one.

This would be so much simpler with associative arrays…

Reply:
I'm not sure why it would be harder with pairs of strings. Your example in the docs could be like
```bash
       LP_MARK_OS=(
           "x86_64" "${BLUE}x64${NO_COL}"
           "i386" "i3"
           "i686" "i6"
           ...
       )
```

The the for loop here would change like 
```diff
-    for key_rep in "${pairs[@]}"; do
-        key="${key_rep%=*}"
+    for (( i = 0; i < ${#pairs[@]}; I+=2 )); do
+        key=${pairs[_LP_FIRST_INDEX+i]}
         if [[ "$key" == "$string" ]]; then
-            rep="${key_rep#*=}"
+            rep=${pairs[_LP_FIRST_INDEX+i+1]-}
```

Reply:
I should have thought about that. I like it better, indeed, it feels more natural than the separator.